tests/lib/Migrations/Version20210913225128.php

```diff
@@ -19,20 +16,81 @@ public function getDescription(): string
 
     public function up(Schema $schema): void
     {
-        $this->addSql('CREATE TABLE `schedule` (id INT AUTO_INCREMENT NOT NULL, PRIMARY KEY(id))');
-        $this->addSql('
-            CREATE TABLE `schedule_rule` (
-                id INT AUTO_INCREMENT NOT NULL, 
-                rules TEXT, 
-                schedule INT FOREIGNKEY schedule_rule_schedule_id_fk 
-                    REFERENCES `schedule`(`id`) ON DELETE CASCADE ON UPDATE RESTRICT, 
-                PRIMARY KEY(id))
-                ');
+        $this->addSql('CREATE TABLE `activity` (
+              id INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
+              `name` VARCHAR(255) NOT NULL,
+              `schedule` INT NOT NULL,
+              `leader` INT DEFAULT NULL,
+              CONSTRAINT activity_leader_id_fk 
+                  FOREIGN KEY (`leader`) REFERENCES leader(`id`),
+              CONSTRAINT activity_schedule_id_fk 
+                  FOREIGN KEY (`schedule`) REFERENCES schedule(`id`) 
+              )');
+
+        $this->addSql('CREATE TABLE `leader` (
+            id INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
+            `name` VARCHAR(255) NOT NULL,
+            `surname` VARCHAR(255) NOT NULL)
+        ');
+
+        $this->addSql('CREATE TABLE `schedule` (
+            id INT AUTO_INCREMENT NOT NULL, PRIMARY KEY(id)
+        )');
+
+        $this->addSql('CREATE TABLE `schedule_rule` (
+          id INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
+            `name` VARCHAR(255) NOT NULL,
+            `rules` INT NOT NULL,
+            `schedule`INT NOT NULL ,
+            `type` INT NOT NULL,
+            CONSTRAINT schedule_rule_schedule_id_fk 
+                FOREIGN KEY (`schedule`) REFERENCES schedule(`id`),
+            CONSTRAINT schedule_rule_schedule_type_id_fk 
+                FOREIGN KEY (`type`) REFERENCES schedule_type(`id`)
+        )');
+
+        $this->addSql('CREATE TABLE `schedule_type` (
+            id INT AUTO_INCREMENT NOT NULL PRIMARY KEY,
+            `name` VARCHAR(255) NOT NULL
+        )');
```

Comment:
What is the purpose of these types? Let's see how can we save the particular schedule to this schema.